node/status_node.go

```diff
@@ -176,7 +177,15 @@ func (n *StatusNode) setupRPCClient() (err error) {
 	return
 }
 
+func (n *StatusNode) discoveryEnabled() bool {
+	return n.config != nil && n.config.NoDiscovery && n.config.ClusterConfig != nil
+}
+
 func (n *StatusNode) startPeerPool() error {
```

Comment:
Rename to `startDiscovery` perhaps? Seems like the scope of the responsibility of this function has expanded and it now effectively start discovery, right?

Also, the code above:

```go
if n.discoveryEnabled() {
    return n.startPeerPool()
}
```

becomes more consistent

```go
if n.discoveryEnabled() {
    return n.startDiscovery()
}
```